src/main/java/domain/AttendanceHistory.java

```diff
@@ -0,0 +1,57 @@
+package domain;
+
+import java.time.LocalDateTime;
+
+public class AttendanceHistory implements Comparable<AttendanceHistory> {
+    private final LocalDateTime attendanceTime;
+    private final AttendanceResult attendanceResult;
+
+    public final static int ABSENT_DEFAULT_HOUR = 23;
+    public final static int ABSENT_DEFAULT_MINUTE = 59;
+    private final static int START_TIME = 8;
+    private final static int END_TIME = 23;
+
+    public AttendanceHistory(LocalDateTime attendanceTime) {
+        validateHistory(attendanceTime);
+        this.attendanceTime = attendanceTime;
+        attendanceResult = getAttendanceResult(attendanceTime);
+    }
+
+    public boolean isBeforeHistory(LocalDateTime time) {
+        LocalDateTime standardTime = LocalDateTime.of(time.getYear(), time.getMonthValue(), time.getDayOfMonth(), 0, 0);
+        return attendanceTime.isBefore(standardTime);
+    }
+
+    @Override
+    public int compareTo(AttendanceHistory o) {
+        return this.attendanceTime.compareTo(o.attendanceTime);
+    }
+
+    private void validateHistory(LocalDateTime attendanceTime) {
+        Holiday.validate(attendanceTime);
+        validateOpeningHours(attendanceTime);
+    }
+
+    private void validateOpeningHours(LocalDateTime attendanceTime) {
+        if (!(attendanceTime.getHour() == ABSENT_DEFAULT_HOUR && attendanceTime.getMinute() == ABSENT_DEFAULT_MINUTE)
+                && (attendanceTime.isBefore(LocalDateTime.of(attendanceTime.getYear(), attendanceTime.getMonthValue(),
+                attendanceTime.getDayOfMonth(), START_TIME, 0)) ||
+                (attendanceTime.isAfter(LocalDateTime.of(attendanceTime.getYear(), attendanceTime.getMonthValue(),
+                        attendanceTime.getDayOfMonth(), END_TIME, 0)))
```

Comment:
시작과 끝 일시 생성을 메서드로 분리해보면 가독성이 좋을 것 같군요 😃 

```suggestion
                && (attendanceTime.isBefore(시작시간(~~)) ||
                (attendanceTime.isAfter(끝시간(~~)))
```

+) LocalTime 으로만 비교해도 될 것 같은데 year, month, day 도 함께 비교하는 이유가 궁금해요!

Reply:
history 자체에서는 LocalDateTime이 저장되기 때문에 해당 메서드에서도 관성적으로 LocalDateTime을 작성했었는데 해당 메서드에는 굳이 LocalDateTime까지 필요가 필요가 없는 게 맞다고 느껴집니다.

따라서 LocalTime으로 바꾸고 isBefore 와 isAfter의 경우 있는줄 모르고 만들려 했는데 이미 존재하길래 해당 메서드를 사용하는 걸로 리팩토링했습니다

Reply:
보통 조건문은 조건이 많아지다 보면 가독성이 많이 낮아진다고 생각해요 :)
그래서 헬퍼 메서드들을 사용해서 가독성을 높일 수 있는지 점검해보는게 좋아요! ㅎㅎ